vanilla/vanilla
# Sort category dropdown list alphabetically

Closes vanilla/support#1941.

Currently, the categories in the dropdown list are sorted according to the "Tree Left" value. In most cases, admins can order categories as they choose. However, when a category is set to display as flat, it isn't possible to change the order of the child categories. In the dashboard, the children categories appear in alphabetical order, but in the category dropdown menu, they aren't sorted alphabetically, which makes it difficult for users to find the category they want. This PR sorts the children categories alphabetically in the dropdown menu by adding the sortFlatCategories() method.

This PR also introduces the method loadAllCategoriesDb(), which is refactored code from loadAllCategories().

### TO TEST
1. Make sure you have one or more categories are set to display "flat" and have multiple child categories.
1. Create a new discussion.
1. Go to select a category.
1. Note that the child categories are not sorted alphabetically.
1. Check out this branch.
1. Refresh the page and note that the child categories are sorted alphabetically.
